### PR TITLE
- added support for pools

### PR DESCRIPTION
**Pull request by aschnell**
Adds support for pools. A pool is a collection of devices, e.g. HDDs or SSDs. The pool provides functions to create several partitions on devices in the pool. This makes it easy to e.g. create the partitions required for a RAID. Also see the documentation in Pool.h and the new integration tests.

Not for any specific PBI, bug or feature request.

### CODE REVIEW

File: storage/Pool.h [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	 * Add a device to the pool. The devicegraph the device belongs to is irrelevant.
+	 *
+	 * @see remove_device(const Device*)
+	 *
+	 * @throw Exception
+	 */
+	void add_device(const Device* device);
+
+	/**
+	 * Remove a device to the pool. The devicegraph the device belongs to is irrelevant.
```
Comment (shundhammer) [inline storage/Pool.h:95]: "Remove a device **from** the pool."

File: storage/Pool.h [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	void remove_device(const Device* device);
+
+	/**
+	 * Get the devices of the pool available in the devicegraph. It is no error if
+	 * devices are not available in the devicegraph.
+	 */
+	std::vector<const Device*> get_devices(const Devicegraph* devicegraph) const;
+
+	/**
+	 * Create number partitions of size in the pool. Devices in the pool not of type
```
Comment (shundhammer) [inline storage/Pool.h:110]: "Create **a** number **of** partitions..."

File: storage/Pool.h [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	 * partitionable or without a partition table are ignored.
+	 *
+	 * The sizes of the created partitions may be different from size due to
+	 * alignment. They may even be pairwise different.
+	 *
+	 * The result is notdeterministic.
```
Comment (shundhammer) [inline storage/Pool.h:116]: "**non**deterministic"

File: storage/StorageImpl.cc [hunk 4]
```diff
@@ -445,4 +443,88 @@ namespace storage
...
+    {
+	ST_CHECK_PTR(devicegraph);
+
+	// TODO more types, e.g. dasd, multipath, pmem, nvme?
+	// TODO check partition table?
+	// TODO do not add already existing devices
+
+	for (const Disk* disk : Disk::get_all(devicegraph))
+	{
+	    string name = disk->is_rotational() ? "HDDs" : "SSDs";
```
Comment (shundhammer) [inline storage/StorageImpl.cc:458]: Will any USB sticks / SD cards that happen to be plugged in also appear in the "SSDs" pool? Or in a separate one? Or in none?

**Comment by shundhammer** [comment]
Just some typos, otherwise LGTM.

File: storage/StorageImpl.cc (hunk 4 above)
Comment (aschnell) [reply storage/StorageImpl.cc:458]: They will also appear as SSDs. What pools should be generated still has to be discussed.

**Pull request merged**